src/radiation/source_terms_multi_group.hpp

```diff
@@ -160,6 +160,89 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::ComputeJacobianForGasAndDustDecouple
 	return result;
 }
 
+// Compute kappaF and the delta_nu_kappa_B_at_edge term. kappaF is used to compute the work term and the delta_nu_kappa_B_at_edge term is used to compute the
+// transport between groups in the momentum function. Only the last two arguments (kappaFVec, delta_nu_kappa_B_at_edge) are modified in this function.
+template <typename problem_t>
+AMREX_GPU_DEVICE void RadSystem<problem_t>::ComputeModelDependentKappaFAndDeltaTerms(
+    double const T, double const rho, amrex::GpuArray<double, nGroups_ + 1> const &rad_boundaries, quokka::valarray<double, nGroups_> const &fourPiBoverC,
+    quokka::valarray<double, nGroups_> const &kappaP, quokka::valarray<double, nGroups_> const &kappaE, quokka::valarray<double, nGroups_> &kappaF,
+    amrex::GpuArray<double, nGroups_> &delta_nu_kappa_B_at_edge)
```

Comment:
This could return a struct instead of having a lot of arguments.

Reply:
I changed it to take `OpacityTerms` struct instead. It will make use of the values of `opacity_terms.kappaP` and `opacity_terms.kappaE` and update `opacity_terms.`kappaF` and `delta_nu_kappa_B_at_edge`. I think this is simpler than creating yet another struct for just two variables.